entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,120 @@
+
+> [!TIP]
+> You can view this article's [sample](https://github.com/aspnet/EntityFramework.Docs/tree/master/samples/core/Querying) on GitHub.
+
+## Join
+
+LINQ Join operator allows you to connect two data sources based on the key selector on each source and generates a tuple of values when key matches. This naturally translates to `INNER JOIN` on relational databases. While LINQ Join has outer & inner key selectors, databases take a condition as join condition and EF Core will generate join condition by comparing outer key selector to inner key selector through equality operation. Further, if key selectors are anonymous types, EF Core will generate join condition to perform equality component wise.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#Join)]
+
+```SQL
+SELECT [p].[PersonId], [p].[Name], [p].[PhotoId], [p0].[PersonPhotoId], [p0].[Caption], [p0].[Photo]
+FROM [PersonPhoto] AS [p0]
+INNER JOIN [Person] AS [p] ON [p0].[PersonPhotoId] = [p].[PhotoId]
+```
+
+## GroupJoin
+
+LINQ GroupJoin operator allows you to connect two data sources in same way as Join but the different here is, it creates a group of values from inner source which matches key with element from outer source. Performing query like following generates result of `Blog` & `IEnumerable<Post>`. Since databases (especially relational databases) do not support representing a collection of client side objects, GroupJoin cannot be translated to server in many cases. It requires to get all the data from server to perform GroupJoin without special selector. But if the selector is limiting data being selected then fetching all data from server may cause performance issues. Hence EF Core does not translate GroupJoin.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupJoin)]
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupJoinComposed)]
+
+## SelectMany
+
+LINQ SelectMany operator allows you to enumerate over collection selector for each element in outer source and generate tuples of value from each data source. In a way it is a join but without any condition so every record is connected with other. Depending on how collection selector is related to outer data source, SelectMany can translate to various different queries on server side.
+
+### Collection selector does not reference outer
+
+When collection selector is not referencing anything from outer source, it is cartesian product of both data sources. In translates to `CROSS JOIN` in relational databases.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#SelectManyConvertedToCrossJoin)]
+
+```SQL
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], [p].[PostId], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+FROM [Blogs] AS [b]
+CROSS JOIN [Posts] AS [p]
+```
+
+### Collection selector references outer in predicate
```

Comment:
Another word that might be too technical: *predicate*

Reply:
where? filter?

Reply:
where clause?